compliance/collection/metrics/metrics.go

```diff
@@ -0,0 +1,141 @@
+package metrics
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/metrics"
+)
+
+var (
+	numberOfRHELPackages = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.ComplianceSubsystem.String(),
+		Name:      "packages_in_inventory",
+		Help:      "Number of packages discovered by the last Node Inventory (per Node)",
+	},
+		[]string{
+			// The Node this scan belongs to
+			"node_name",
+			// The OS name and version of the Node
+			"os_namespace",
+		})
+
+	numberOfContentSets = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.ComplianceSubsystem.String(),
+		Name:      "content_sets_in_inventory",
+		Help:      "Number of content sets discovered by the last Node Inventory (per Node)",
+	},
+		[]string{
+			// The Node this scan belongs to
+			"node_name",
+			// The OS name and version of the Node
+			"os_namespace",
+		})
+
+	scanDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.ComplianceSubsystem.String(),
+		Name:      "inventory_scan_duration_seconds",
+		Help:      "Scan duration for Node Inventory (per Node) in seconds",
+	},
+		[]string{
+			// The Node this scan belongs to
+			"node_name",
+			// Whether the inventory run was completed successfully
+			"error",
+		})
+
+	rescanInterval = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.ComplianceSubsystem.String(),
+		Name:      "rescan_interval_seconds",
+		Help:      "Time in seconds between Node Inventory runs",
+	},
+		[]string{
+			// The Node this scan belongs to
+			"node_name",
+		})
+
+	protobufMessageSize = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.ComplianceSubsystem.String(),
+		Name:      "protobuf_inventory_message_size",
```

Comment:
Shall we add unit name as postfix to the name?

Reply:
We need to configure histogram buckets here as 100% of observations fall into the `le="+Inf"` bucket.
I would propose the following buckets: 500B, 1kB, 10kB, 50kB, 100kB, 500kB, 1000kB,...

Reply:
Thanks for the insight w.r.t. bucket sizes. This is one of the things I wanted to test experimentally.
Now that we know, I have added your proposed buckets. 
I also configured buckets for the scan time hist to be `[]float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 20}`, as I expect the scan time to be a bit higher on production nodes that are running workloads.

Reply:
That is fine, but note when we have scan times above 20, we will be blind how large they are as they fall into the bucket of `20 to +Inf`. I would suggest to always include some exponential buckets at the end, for example: `[]float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 20, 50, 100, 500, 1000}`

Reply:
That's a great practice to stick to, let me update